ksql-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java

```diff
@@ -71,7 +71,12 @@ LogicalSchema schema() {
     awaitRunning();
 
     try {
-      return kafkaStreams.store(stateStoreName, queryableStoreType);
+      if (ksqlConfig.getBoolean(KsqlConfig.KSQL_PULL_QUERIES_ENABLE_CONFIG)) {
```

Comment:
Do you mean `KsqlConfig.KSQL_QUERY_PULL_ALLOW_STALE_READS`?

Simplify to `return kafkaStreams.store(stateStoreName, queryableStoreType, 
    ksqlConfig.getBoolean(KsqlConfig.KSQL_PULL_QUERIES_ENABLE_CONFIG))`?

Reply:
Yes! Good catch! Coding with fever does not help :)